sched/task/task_vfork.c

```diff
@@ -151,6 +152,14 @@ FAR struct task_tcb_s *nxtask_setup_vfork(start_t retaddr)
       goto errout_with_tcb;
     }
 
+  /* Duplicate the parent tasks environment */
+
+  ret = group_setupenv(child, group_getenv());
```

Comment:
is it too early to call group_setupenv here?

Reply:
No, group_getenv() will return the parent process environment which is duplicatead as is. vfork shares the same address environment for the child so this is correct.

Reply:
> No, group_getenv() will return the parent process environment which is duplicatead as is. vfork shares the same address environment for the child so this is correct.
> 

Yes, you are right.

> On the other hand, I don't know if this does _what it is supposed to do_. The environment handling might not be really compatible for vfork() / execve(), and needs the modification to pass envp like argv.

I think so, how about to support execve?

Reply:
I need execve() anyway, but it will take some time to implement. I can live with posix_spawn() for now.